### PR TITLE
Magento_ConfigurableProduct: avoid using deprecated escape* methods f…

### PR DESCRIPTION
**Pull request by sergiynezbritskiy**
### Description (*)

Avoid using deprecated escape* methods from AbstractBlock

### Related Pull Requests

https://github.com/magento/magento2/pull/31610

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/composite/fieldset/configurable.phtml [hunk 1]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Composite\Fieldset\Configurable */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Composite\Fieldset\Configurable
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/composite/fieldset/configurable.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Composite\Fieldset\Configurable $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/attributes_values.phtml [hunk 2]
```diff
@@ -4,17 +4,20 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\AttributeValues */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\AttributeValues
+ * @var \Magento\Framework\Escaper $escaper
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/attributes_values.phtml]:
```suggestion
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\AttributeValues $block
 * @var \Magento\Framework\Escaper $escaper
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/bulk.phtml [hunk 3]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Bulk */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Bulk
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/bulk.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Bulk $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/select_attributes.phtml [hunk 4]
```diff
@@ -4,19 +4,22 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\SelectAttributes */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\SelectAttributes
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/select_attributes.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\SelectAttributes $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/summary.phtml [hunk 5]
```diff
@@ -4,10 +4,13 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Summary */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Summary
+ * @var \Magento\Framework\Escaper $escaper
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/summary.phtml]:
```suggestion
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Steps\Summary $block
 * @var \Magento\Framework\Escaper $escaper
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/config.phtml [hunk 6]
```diff
@@ -4,22 +4,25 @@
  * See COPYING.txt for license details.
  */
 
-/** @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config
+ * @var \Magento\Framework\Escaper $escaper
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/config.phtml]:
```suggestion
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config $block
 * @var \Magento\Framework\Escaper $escaper
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/matrix.phtml [hunk 7]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/** @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/matrix.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/wizard.phtml [hunk 8]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/** @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/wizard.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/product/configurable/affected-attribute-set-selector/js.phtml [hunk 9]
```diff
@@ -4,7 +4,10 @@
  * See COPYING.txt for license details.
  */
 
-/* @var $block \Magento\ConfigurableProduct\Block\Product\Configurable\AttributeSelector */
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Product\Configurable\AttributeSelector
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/adminhtml/templates/product/configurable/affected-attribute-set-selector/js.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Block\Product\Configurable\AttributeSelector $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer
 */
```

File: app/code/Magento/ConfigurableProduct/view/base/templates/product/price/final_price.phtml [hunk 10]
```diff
@@ -4,7 +4,10 @@
...
  */
 
-/** @var \Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox$block */
+/**
+ * @var \Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox $block
+ * @var \Magento\Framework\Escaper $escaper
+ */
 /** @var \Magento\Framework\Pricing\Price\PriceInterface $priceModel */
 $priceModel = $block->getPriceType('regular_price');
 /** @var \Magento\Framework\Pricing\Price\PriceInterface $finalPriceModel */
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/base/templates/product/price/final_price.phtml]:
```suggestion
/**
 * @var \Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox $block
 * @var \Magento\Framework\Escaper $escaper
 * @var \Magento\Framework\Pricing\Price\PriceInterface $priceModel
 * @var \Magento\Framework\Pricing\Price\PriceInterface $finalPriceModel
 */
$priceModel = $block->getPriceType('regular_price');
```

File: app/code/Magento/ConfigurableProduct/view/frontend/templates/product/view/type/options/configurable.phtml [hunk 11]
```diff
@@ -6,23 +6,26 @@
 ?>
 
 <?php
-/** @var $block \Magento\ConfigurableProduct\Block\Product\View\Type\Configurable*/
+/**
+ * @var $block \Magento\ConfigurableProduct\Block\Product\View\Type\Configurable
+ * @var \Magento\Framework\Escaper $escaper
```
Comment (ihor-sviziev) [inline app/code/Magento/ConfigurableProduct/view/frontend/templates/product/view/type/options/configurable.phtml]:
```suggestion
 * @var \Magento\ConfigurableProduct\Block\Product\View\Type\Configurable $block
 * @var \Magento\Framework\Escaper $escaper
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/ConfigurableProduct/view/frontend/templates/product/view/type/options/configurable.phtml (hunk 11 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/frontend/templates/product/view/type/options/configurable.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/base/templates/product/price/final_price.phtml (hunk 10 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/base/templates/product/price/final_price.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/product/configurable/affected-attribute-set-selector/js.phtml (hunk 9 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/product/configurable/affected-attribute-set-selector/js.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/wizard.phtml (hunk 8 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/wizard.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/matrix.phtml (hunk 7 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/matrix.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/composite/fieldset/configurable.phtml (hunk 1 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/composite/fieldset/configurable.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/attributes_values.phtml (hunk 2 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/attributes_values.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/bulk.phtml (hunk 3 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/bulk.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/select_attributes.phtml (hunk 4 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/select_attributes.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/summary.phtml (hunk 5 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/attribute/steps/summary.phtml]: fixed

File: app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/config.phtml (hunk 6 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/ConfigurableProduct/view/adminhtml/templates/catalog/product/edit/super/config.phtml]: fixed

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Database Compare

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Database Compare

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes form this PR.

**Pull request closed without merging**